### PR TITLE
Update patch releasenotes format

### PR DESCRIPTION
**Pull request by hanxiaop**
Using the generated template has a lint error: https://prow.istio.io/view/gs/istio-prow/pr-logs/pull/istio_istio.io/14301/lint_istio.io/1734045878638874624

### CODE REVIEW

**Comment by hanxiaop** [comment]
Lint tool in `tools` and `istio.io` behavior differently.

**Comment by windsonsea** [comment]
Seems the change is correct.

**Comment by hanxiaop** [comment]
Without this change, we (RMs) would have to add an extra hashtag '#' to the header each time we generate a new patch release note. I'm not sure why the linter behaves differently here compared to the doc site.

**Comment by hanxiaop** [comment]
@ericvn Should we consider excluding files in `cmd/gen-release-notes/templates` from linting? These files will actually be provided in a separate PR on the istio.io repo, where they will be performed with the doc lint test. Or do you have any better solutions?

**Review by howardjohn — APPROVED** [review]

File: Makefile.core.mk [hunk 1]
```diff
@@ -14,6 +14,9 @@
 
 SHELL := /bin/bash -o pipefail
 
+FINDFILES=find . \( -path ./common-protos -o -path ./.git -o -path ./out -o -path ./.github -o -path ./licenses -o -path ./vendor -o -path ./cmd/gen-release-notes/templates \) -prune -o -type f
```
Comment (ericvn) [inline Makefile.core.mk:17]: I made a comment in https://github.com/istio/common-files/pull/960#issuecomment-1852350517 where we could simply add an additional variable here, like FINDFILES_IGNORE and then add that to the find command in common-files.

Comment (hanxiaop) [reply Makefile.core.mk:17]: Makes sense, thanks

**Review by ericvn — APPROVED** [review]
Thanks for making the small change.

**Pull request merged**